File: 0035-search-insert-position/35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python3
#
# [35] Search Insert Position
#

# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)

        while left < right:
            mid = (left + right) // 2

            if target == nums[mid]:
                return mid
            elif target > nums[mid]:
                left = mid + 1
            else:
                right = mid
                
        return left
# @lc code=end

File: 0035-search-insert-position/35.search-insert-position.rs
/*
 * @lc app=leetcode id=35 lang=rust
 *
 * [35] Search Insert Position
 */

// @lc code=start
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut left : usize = 0;
        let mut right = nums.len();

        while left < right {
            let mut mid = ((left + right) as f32 / 2.0).floor() as usize;

            if nums[mid] == target {
                return mid as i32;
            } else if nums[mid] < target {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return left as i32;
    }
}
// @lc code=end

